ttnn/cpp/ttnn/operations/bernoulli/device/kernels/writer_bernoulli.cpp

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "common/constants.hpp"
+#include "dataflow_api.h"
+
+using namespace tt;
+
+void kernel_main() {
+    constexpr uint32_t in_cb_id = get_compile_time_arg_val(0);
+    constexpr uint32_t intermed_cb_id = get_compile_time_arg_val(1);
+    constexpr uint32_t intermed1_cb_id = get_compile_time_arg_val(2);
+    constexpr bool output_is_dram = get_compile_time_arg_val(3) == 1;
+
+    uint32_t out_addr = get_arg_val<uint32_t>(0);
+    uint32_t start_id = get_arg_val<uint32_t>(1);
+    uint32_t num_tiles = get_arg_val<uint32_t>(2);
+    uint32_t end_id = start_id + num_tiles;
+
+    const InterleavedAddrGenFast<output_is_dram> output_addrg = {
+        .bank_base_address = out_addr,
+        .page_size = get_tile_size(intermed1_cb_id),
+        .data_format = get_dataformat(intermed1_cb_id)};
+
+    uint32_t max_int = 2147483647;
+
+    cb_reserve_back(intermed1_cb_id, 1);
+    uint32_t intermed1_cb_write_ptr = get_write_ptr(intermed1_cb_id);
+
+    for (uint32_t i = start_id; i < end_id; ++i) {
+        cb_wait_front(in_cb_id, 1);
+        cb_wait_front(intermed_cb_id, 1);
+
+        uint32_t intermed_cb_read_ptr = get_read_ptr(intermed_cb_id);
+        uint32_t in_cb_read_ptr = get_read_ptr(in_cb_id);
+
+        uint8_t *in_cb_addr = reinterpret_cast<uint8_t *>(in_cb_read_ptr);
+        uint32_t *intermed_cb_addr = reinterpret_cast<uint32_t *>(intermed_cb_read_ptr);
+        uint8_t *intermed1_cb_addr = reinterpret_cast<uint8_t *>(intermed1_cb_write_ptr);
+
+        for (uint32_t k = 0; k < constants::TILE_WIDTH; k++) {
+            for (uint32_t j = 0; j < constants::TILE_HEIGHT; j++) {
+                uint32_t rand_uint32 = *intermed_cb_addr;
+                //  hardware PRNG is not uniformly distribute.
+                // Generated rand_floats in range [0, max_uint32 / 2] has higher ratio compared to (max_uint32 / 2,
+                // max_uint32), which makes the output tensor contains more 1 value than 0. I divide rand_float with
+                // max_int instead of max_uint to balance the number of 1 and 0 value in output tensor.
+                float rand_float = static_cast<float>(rand_uint32) / max_int;
```

Comment:
This comment should be changed to align with the fact you use max_int instead of max_uint32.

Reply:
Updated.